### PR TITLE
On JDK 8, output directory should be added to the classpath

### LINKED ISSUE #122: Plugin doesn't work with JDK 8 projects
A regression was introduced in #91 and the plugin no longer works with JDK 8 projects.

On running the plugin on JDK 8, the classpath is no longer populated with the path to `target/classes` directory.

This is a high priority issue and its resolution should be followed by a release.

### PR DESCRIPTION
**Pull request by abhinayagarwal**
Fix #122

### CODE REVIEW

File: src/main/java/org/openjfx/JavaFXRunMojo.java [hunk 1]
```diff
@@ -145,7 +145,7 @@ public void execute() throws MojoExecutionException {
             }
         }
 
-        if (classpathElements != null && !classpathElements.isEmpty()) {
+        if (classpathElements != null && (oldJDK || !classpathElements.isEmpty())) {
```
Comment (jperedadnr) [inline src/main/java/org/openjfx/JavaFXRunMojo.java:148]: Feels odd, maybe
```
oldJDK  || (classpathElements != null && !classpathElements.isEmpty())
```
?
I take that `classpathElements` is never null, but the first condition should happen also if it was?
Then maybe:
```
if (oldJDK || (classpathElements != null  && !classpathElements.isEmpty())) {
            commandArguments.add("-classpath");
            String classpath = "";
            if (oldJDK || runtimePathOption == CLASSPATH) {
                classpath = project.getBuild().getOutputDirectory() + File.pathSeparator;
            }
if (classpathElements != null) {
            classpath += StringUtils.join(classpathElements.iterator(), File.pathSeparator);
}
            commandArguments.add(classpath);
        }
```

Comment (abhinayagarwal) [reply src/main/java/org/openjfx/JavaFXRunMojo.java:148]: `classpathElements` is initialized inside `preparePaths()` which is called a few lines above this condition. Given `classpathElements` is always initialized, we can remove the null check and simplify the condition to:

```

if (oldJDK || !classpathElements.isEmpty()) {
    commandArguments.add("-classpath");
    String classpath = "";
    if (oldJDK || runtimePathOption == CLASSPATH) {
        classpath = project.getBuild().getOutputDirectory() + File.pathSeparator;
    }
    classpath += StringUtils.join(classpathElements.iterator(), File.pathSeparator);
    commandArguments.add(classpath);
}
```

Comment (jperedadnr) [reply src/main/java/org/openjfx/JavaFXRunMojo.java:148]: Makes sense.

**Review by jperedadnr — APPROVED** [review]

**Review by jperedadnr — COMMENTED** [review]
Tests are failing, `classpathElements != null` was needed after all.

**Comment by abhinayagarwal** [comment]
@jperedadnr Weird, seems like something sketchy going on in the tests. For now, I have reverted the change.

**Review by jperedadnr — APPROVED** [review]

**Pull request merged**